### PR TITLE
Fix s390x Warning: File is not PE

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

#16168. It works, but not pretty. pe.c is already big. Any suggestion where to put these code, or another approach to deal with endian?

**Test plan**

https://travis-ci.com/github/eagleoflqj/radare2/jobs/443283549 no such warnings, and db/formats/pe/dllnullep fixed.

**Closing issues**

...

### CODE REVIEW

File: libr/bin/format/pe/pe.c [hunk 1]
```diff
@@ -574,12 +574,113 @@ static int bin_pe_parse_imports(struct PE_(r_bin_pe_obj_t)* bin,
 	return false;
 }
 
+#if R_SYS_ENDIAN
+static void PE_(image_dos_header_swapendian)(PE_(image_dos_header) *header) {
+#define swapendian(field) r_mem_swapendian ((ut8*) &header->field, (ut8*) &header->field, sizeof (header->field))
```
Comment (trufae) [inline libr/bin/format/pe/pe.c:579]: please use r_read APIs, which take care of this depending on R_SYS_ENDIAN

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/bin/format/pe/pe.c (hunk 1 above)
Comment (eagleoflqj) [reply libr/bin/format/pe/pe.c:579]: I guess you mean r_buf_read_lexx since we are dealing with RBuffer instead of ut8 array.

**Review by XVilka — APPROVED** [review]

File: libr/bin/format/pe/pe.c [hunk 2]
```diff
@@ -597,7 +697,7 @@ static int bin_pe_init_hdr(struct PE_(r_bin_pe_obj_t)* bin) {
 		return false;
 	}
 	bin->nt_header_offset = bin->dos_header->e_lfanew;
-	if (r_buf_read_at (bin->b, bin->dos_header->e_lfanew, (ut8*) bin->nt_headers, sizeof (PE_(image_nt_headers))) < 0) {
+	if (read_nt_headers (bin->b, bin->dos_header->e_lfanew, bin->nt_headers) < 0) {
 		bprintf ("Warning: read (dos header)\n");
```
Comment (wargio) [inline libr/bin/format/pe/pe.c:701]: shouldn't this be nt header ? dos header is above.

**Review by wargio — CHANGES_REQUESTED** [review]
besides the stupid warning, i love this.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Review by trufae — APPROVED** [review]

**Pull request merged**